### DISCUSSION TITLE
Issue building 8.6.1 on Mac - Latest commit builds though

### DISCUSSION THREAD

**Discussion by ciaransweet** [esmf-org/esmf#287] [Building / Installing] [answered]
### Requirements

- [X] Reviewed [ESMF User's Guide](https://earthsystemmodeling.org/doc/)
- [X] Searched [GitHub Discussions](https://github.com/orgs/esmf-org/discussions?discussions_q=)

### Affiliation(s)

Development Seed

### ESMF Version

v8.6.1

### Issue

Hey folks,

I'm ending up building ESMF as I don't use conda so installing `esmpy` this via pip + the repository cloned locally.

I managed to run `ESMF_DIR=$(pwd) make` and successfully built the library with this commit: https://github.com/esmf-org/esmf/commit/82ba326579f89b8946c99b2f953d37223b28d7e0

However, noticing that when I tried to import `esmpy` I had a version mismatch, I checked out this commit: https://github.com/esmf-org/esmf/releases/tag/v8.6.1

Unfortunately I now get this error:

```
gfortran -c -O -fPIC  -pthread -frecursive -ffree-line-length-none -fno-backtrace -I/Users/ciaran/dev/esmf/src/addon/NUOPC/src -I/Users/ciaran/dev/esmf/build_config/Darwin.gfortranclang.default -I/Users/ciaran/dev/esmf/src/Infrastructure -I/Users/ciaran/dev/esmf/src/Superstructure -I/Users/ciaran/dev/esmf/mod/modO/Darwin.gfortranclang.64.mpiuni.default -I/Users/ciaran/dev/esmf/src/include   -DESMF_NO_INTEGER_1_BYTE -DESMF_NO_INTEGER_2_BYTE -DESMF_VERSION_STRING_GIT='"v8.6.1"' -DESMF_MPIUNI -I/Users/ciaran/dev/esmf/src/Infrastructure/stubs/mpiuni -DESMF_LOWERCASE_SINGLEUNDERSCORE -DESMF_MOAB=1 -DESMF_LAPACK=1 -DESMF_LAPACK_INTERNAL=1 -DESMF_NO_ACC_SOFTWARE_STACK=1 -DESMF_YAMLCPP=1 -DESMF_YAML=1 -DESMF_NO_OPENMP -DESMF_NO_OPENACC -DESMF_BOPT_O -DESMF_TESTCOMPTUNNEL -DS64=1 -DESMF_OS_Darwin=1 -DESMF_COMM=mpiuni -DESMF_DIR=/Users/ciaran/dev/esmf /Users/ciaran/dev/esmf/src/addon/NUOPC/src/NUOPC_Base.F90 -o /Users/ciaran/dev/esmf/obj/objO/Darwin.gfortranclang.64.mpiuni.default/NUOPC_Base.o
/Users/ciaran/dev/esmf/src/addon/NUOPC/src/NUOPC_Base.F90:23:7:

   23 |   use NUOPC_FieldDictionaryApi
      |       1
Fatal Error: Mismatch in components of derived type '__vtype_esmf_attributemod_Esmf_attpack' from 'esmf_attributemod' at (1): expecting 'getpayload', but got 'initialize'
compilation terminated.
make[8]: *** [/Users/ciaran/dev/esmf/obj/objO/Darwin.gfortranclang.64.mpiuni.default/NUOPC_Base.o] Error 1
make[7]: *** [esmflib] Error 2
make[6]: *** [tree_lib-default] Error 2
make[5]: *** [tree] Error 1
make[4]: *** [tree] Error 1
make[3]: *** [tree] Error 1
make[2]: *** [tree] Error 1
make[1]: *** [build_libs] Error 2
make: *** [lib] Error 2
```

Any ideas? Thanks!

### Autotag

@oehmke

**Comment by danrosen25** [comment, marked as answer]
For small esmpy project you can use this Docker image: https://hub.docker.com/repository/docker/esmf/esmf-build-release/general 

The pip installation of esmpy is working for me with the latest develop code
```
cd $ESMF_DIR/src/addon/esmpy
python3 -m pip install .
```
```
Successfully built esmpy
Installing collected packages: esmpy
  Attempting uninstall: esmpy
    Found existing installation: esmpy 8.7.0b13
    Uninstalling esmpy-8.7.0b13:
      Successfully uninstalled esmpy-8.7.0b13
Successfully installed esmpy-8.7.0b13
```
but you will see a **warning** because this is a beta version
`VersionWarning: ESMF installation version 8.7.0 beta snapshot, ESMPy version 8.7.0b13`
If there's an error then `ESMFMKFILE` in your environment variables hasn't been set correctly and it's pointing to a version of the ESMF library that doesn't match the esmpy installation. Set `ESMFMKFILE` in your environment to the location of the `esmf.mk` file in the ESMF installation.
`esmpy.util.exceptions.VersionMismatch: ESMF installation version 8.6.1, ESMPy version 8.7.0b13`

If you're rebuilding ESMF from the same directory then run `make distclean` before `make`.
